Lib/Src/CUartCom.cpp

```diff
@@ -151,7 +166,7 @@ bool CUartCom::transmit()
         m_uart_de_pin.set(true);
 
         updateTxBuffer();
-        HAL_UART_Transmit_IT(mp_huart, (uint8_t *)m_tx_buffer, m_tx_msg_length);
+        HAL_UART_Transmit_IT(mp_huart, &m_tx_char, BYTE);
```

Comment:
Much clearer is you explicitly state that you are sending 1 of `m_tx_char`. The rules about "magic numbers" only applies when it is not clear where the number is coming from and giving it a name is meant to fix the ambiguity. It does not apply here.
```
HAL_UART_Transmit_IT(mp_huart, &m_tx_char, 1);
```